_posts/2021-07-07-cockpit-248.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Cockpit 248
+author: garrett
+date: '2021-07-07'
+tags: cockpit, machines, ostree
+slug: cockpit-248
+category: release
+summary: ''
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 248, Cockpit-Machines 247, and Cockpit-OSTree 187:
+
+
+## Metrics: Install missing packages
+
+[Cockpit 247](https://cockpit-project.org/blog/cockpit-247.html) introduced setting up a machine for Grafana metrics. However, this only worked if the `pcp` and `redis` packages were already installed. With this Cockpit version, the options will always work, and missing packages get installed as needed.
+
+![screenshot of install missing packages]({{ site.baseurl }}/images/248-metrics-install-missing-packages.png)
+
+![screenshot of install missing packages]({{ site.baseurl }}/images/248-metrics-install-missing-packages-2.png)
+
+This completes the story of [setting up PCP and Grafana metrics with Cockpit](https://cockpit-project.org/blog/pcp-grafana.html), have a look at that blog post for all the details.
+
+## PAM: Deprecate `pam_cockpit_cert` module
+
+The `pam_cockpit_cert` module is no longer required for client certificate authentication, and has been deprecated.  This release continues to ship a stub module which returns `PAM_IGNORE`, but this stub will be dropped in a future release.  If you modified  `/etc/pam.d/cockpit`, the package upgrade will not automatically update the file -- please remove the relevant entry manually in this case.
+
+## Machines: Bug fixes and improvements
+
+Cockpit-Machines has a new release with several bug fixes and minor improvements.
+
+FIXME: Format these BZ links better
+bugzilla.redhat.com/show_bug.cgi?id=1978206
+bugzilla.redhat.com/show_bug.cgi?id=1977810
+bugzilla.redhat.com/show_bug.cgi?id=1934323
+bugzilla.redhat.com/show_bug.cgi?id=1977669
+bugzilla.redhat.com/show_bug.cgi?id=1947241
+bugzilla.redhat.com/show_bug.cgi?id=1974216
+
+## OSTree: Bump epoch to 1
+
+The cockpit-ostree RPM package is being released as version `1:186`, with a newly introduced epoch.  This will re-enable upgrading from an unintentional "version 999" of cockpit-ostree which was accidentally published while testing the release infrastructure.  Packages for Debian-based distributions are not impacted by this change.
+
+## Developers: `main` branch switch
+
+Various Cockpit sub-projects have been switching their main branches to `main` over the past few months. We're happy to announce that the all Cockpit repositories, including the main Cockpit repository, have now finished the switch to using a `main` branch for git.
+
```

Comment:
This set of releases also marks the end of our transition to Dart Sass and Webpack 5, as well as cleanups and modernization to cockpit's build system.